server/sonar-db-dao/src/main/java/org/sonar/db/measure/MeasureDao.java

```diff
@@ -115,6 +115,10 @@
       ids -> mapper(dbSession).selectProjectMeasuresOfDeveloper(developerId, metricIds));
   }
 
+  public List<MeasureDto> selectByComponentsAndMetrics(DbSession dbSession, List<String> componentUuids, List<Integer> metricIds) {
+    return executeLargeInputs(componentUuids, components -> mapper(dbSession).selectByComponentsAndProjectsAndMetrics(components, metricIds));
```

Comment:
remark: I prefer to replace `components` by `c` because you already have `componentUuids` and using components might be confusing. Otherwise you could just name it `partition`, `list`